src/app.ts

```diff
@@ -13,14 +13,64 @@ interface TemperatureSummary {
   average: number
 }
 
+interface FullTemperatureSummary extends TemperatureSummary {
+  amountOfReadings: number;
+}
+
+interface ComputedTemperatureSummaries {
+  [city: string]: {
+    [time: string]: FullTemperatureSummary;
+  }
+}
+
+const summaries: ComputedTemperatureSummaries = {};
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  //add here your code
+  //compute temperature summaries, if there is new reading it updates or creates its summary
+
+  readings.forEach(reading => {
+    
+    const time = reading.time.toString();
+    const temperature = reading.temperature;
+    const city = reading.city;    
+
+    if (city in summaries && time in summaries[city]) {// update summary
+      const summary = summaries[city][time];
+      summary.last = temperature;
+      summary.high = Math.max(temperature, summary.high);
+      summary.low = Math.min(temperature, summary.low);
+      summary.average =
+        (summary.average * summary.amountOfReadings + temperature ) / (summary.amountOfReadings + 1);
+      summary.amountOfReadings += 1;
+    }
+    else {//create summary
+      if (!(city in summaries)){
+        summaries[city] = {};
+      }
+      const summary: FullTemperatureSummary = {
```

Comment:
type is not needed here

Reply:
yes, the structure type checking does the work